### ISSUE TITLE
Only the first message on event NewMessage (for forward a few messages, or for album)

### ISSUE THREAD

**Issue by uzlov** [LonamiWebs/Telethon#1676]
**Checklist**
* [X] The error is in the library's code, and not in my own. (or mb the changes in API of Telegram?)
* [X] I have searched for this issue before posting it and there isn't a duplicate.
* [X] I ran `pip install -U https://github.com/LonamiWebs/Telethon/archive/master.zip` and triggered the bug in the latest version.

**Code that causes the issue**
```python
from telethon.sync import TelegramClient
...
    @client.on(events.NewMessage(chats=input_channels_entities))
    async def handler(event):
        logging.info(f"{event}")
```

The issue was reproduced for new telegram account.
The issue is reproducing not for all accounts (I have one account without it issue).

**Comment by L8KKqTNnge** [comment]
Same! I opened another issue for an album. I think these two might be connected somehow. It worked fine like 2 days ago, but now fails

**Comment by uzlov** [comment]
> It worked fine like 2 days ago, but now fails
the same, issues is appeared 18 Jan

**Comment by Lonami** [comment]
If any of you want to investigate, register a `events.Raw` handler and print the incoming updates, see if the entire album arrives. If it doesn't, Telegram may not be sending the complete album.

For the record, @L8KKqTNnge's issue is #1677.

**Comment by uzlov** [comment]
issue not only for album
I have issue for forward a few messages

more details after tests
> The issue was reproduced for new telegram account.
The issue is reproducing not for all accounts (I have one account without it issue).

I guess it's wrong
the issues for any account
but, still  NO issues for desktop client (ubuntu)
from mobile - 1 message from a few forwarded messages, and 1 message from album

**Comment by uzlov** [comment]
> events.Raw

did the tests
I got
- only one 1 message (media) for the forwarded 12 messages (2 albums + 1 text)
- only one 1 message for the sent album (4 media)

**Comment by uzlov** [comment]
> update

account which has no issues (has all messages), it's account which is used for start telethon
all other accounts - have only 1 message for a few forwarder posts, and for album

**Comment by Lonami** [comment]
Well, if `events.Raw` is not printing all updates, it's likely Telegram isn't sending them at all.

**Comment by uzlov** [comment]
@Lonami Any ideas? Can it relate to any permissions? For example, number of  messages in one time or something like it?

**Comment by uzlov** [comment]
mb it's related https://github.com/tdlib/td/commit/e72c129f9ce3b7463c7a42406aab0d3ebffdc6be

**Comment by Lonami** [comment]
Could you elaborate how it's related?

**Comment by uzlov** [comment]
I mean, can something is changed in API of telegram?
the issue is faced first time ~15.00 18 Jan

**Comment by uzlov** [comment]
@Lonami do you need the steps for reproduce issue?

**Comment by SharkLabDev** [comment]
Have news?

**Comment by Lonami** [comment]
This is likely a server issue. Even Telegram Desktop struggles to print it out correctly, in a megagroup:

<details><summary>Main account screenshot</summary>

![image](https://user-images.githubusercontent.com/6297805/105616894-62656d00-5dda-11eb-9267-d19d8e36c868.png)

</details>

Note the order should've been `1, 2, 3`, but instead `1, 3, 2` appears (and it appeared slowly). It looks okay from the alternative account, which forwarded the messages (in the first screenshot as well, when self-forwarding there are no issues):

<details><summary>Alternative account screenshot</summary>

![image](https://user-images.githubusercontent.com/6297805/105616892-57124180-5dda-11eb-97e5-4d980b75aa09.png)

</details>

Given an alternative account sending the messages "a", "b", then forwarding main account's "1", "2", "3" all at the same time, and then sending "c", "d" produces the following updates from Telegram:

<details><summary>Censored and trimmed updates log</summary>

```python
Updates(
    updates=[
        UpdateNewChannelMessage(
            message=Message(
                id=148,
                peer_id=PeerChannel(
                    channel_id=...
                ),
                date=datetime.datetime(2021, 1, 23, 23, 25, 54, tzinfo=datetime.timezone.utc),
                message='a',
                from_id=PeerUser(
                    user_id=...
                ),
                entities=[
                ],
                replies=MessageReplies(
                    replies=0,
                    replies_pts=167,
                    comments=False,
                    recent_repliers=[
                    ],
                    channel_id=None,
                    max_id=None,
                    read_max_id=None
                ),
            ),
            pts=167,
            pts_count=1
        ),
    ],
    users=[
        User(
            id=...,
            is_self=False,
            contact=True,
            mutual_contact=True,
        ),
    ],
    chats=[
        Channel(
            id=1360765426,
            title='TEST CHANNEL'
            date=datetime.datetime(2021, 1, 2, 11, 22, 59, tzinfo=datetime.timezone.utc),
            creator=True,
            broadcast=False,
            megagroup=True,
        ),
    ],
    date=datetime.datetime(2021, 1, 23, 23, 25, 53, tzinfo=datetime.timezone.utc),
    seq=0
)
148 a
Updates(
    updates=[
        UpdateNewChannelMessage(
            message=Message(
                id=149,
                peer_id=PeerChannel(
                    channel_id=...
                ),
                date=datetime.datetime(2021, 1, 23, 23, 25, 54, tzinfo=datetime.timezone.utc),
                message='b',
                from_id=PeerUser(
                    user_id=...
                ),
                entities=[
                ],
                replies=MessageReplies(
                    replies=0,
                    replies_pts=168,
                    comments=False,
                    recent_repliers=[
                    ],
                    channel_id=None,
                    max_id=None,
                    read_max_id=None
                ),
            ),
            pts=168,
            pts_count=1
        ),
    ],
    date=datetime.datetime(2021, 1, 23, 23, 25, 53, tzinfo=datetime.timezone.utc),
    seq=0
)
149 b
Updates(
    updates=[
        UpdateNewChannelMessage(
            message=Message(
                id=150,
                peer_id=PeerChannel(
                    channel_id=...
                ),
                date=datetime.datetime(2021, 1, 23, 23, 25, 59, tzinfo=datetime.timezone.utc),
                message='1',
                from_id=PeerUser(
                    user_id=...
                ),
                    date=datetime.datetime(2021, 1, 23, 23, 20, 20, tzinfo=datetime.timezone.utc),
                    from_id=PeerUser(
                        user_id=10885151
                    ),
                    from_name=None,
                    channel_post=None,
                    saved_from_peer=None,
                    saved_from_msg_id=None,
                    psa_type=None
                ),
                entities=[
                ],
                replies=MessageReplies(
                    replies=0,
                    replies_pts=169,
                    comments=False,
                    recent_repliers=[
                    ],
                    channel_id=None,
                    max_id=None,
                    read_max_id=None
                ),
            ),
            pts=169,
            pts_count=1
        ),
    ],
    date=datetime.datetime(2021, 1, 23, 23, 25, 58, tzinfo=datetime.timezone.utc),
    seq=0
)
150 1
UpdateShort(
    update=UpdateChannelUserTyping(
        channel_id=...,
        user_id=...,
        action=SendMessageTypingAction(
        ),
        top_msg_id=None
    ),
    date=datetime.datetime(2021, 1, 23, 23, 25, 58, tzinfo=datetime.timezone.utc)
)
Updates(
    updates=[
        UpdateNewChannelMessage(
            message=Message(
                id=153,
                peer_id=PeerChannel(
                    channel_id=...
                ),
                date=datetime.datetime(2021, 1, 23, 23, 26, tzinfo=datetime.timezone.utc),
                message='c',
                from_id=PeerUser(
                    user_id=...
                ),
                entities=[
                ],
                replies=MessageReplies(
                    replies=0,
                    replies_pts=172,
                    comments=False,
                    recent_repliers=[
                    ],
                    channel_id=None,
                    max_id=None,
                    read_max_id=None
                ),
            ),
            pts=172,
            pts_count=1
        ),
    ],
    date=datetime.datetime(2021, 1, 23, 23, 25, 59, tzinfo=datetime.timezone.utc),
    seq=0
)
153 c
Updates(
    updates=[
        UpdateNewChannelMessage(
            message=Message(
                id=154,
                peer_id=PeerChannel(
                    channel_id=...
                ),
                date=datetime.datetime(2021, 1, 23, 23, 26, tzinfo=datetime.timezone.utc),
                message='d',
                from_id=PeerUser(
                    user_id=...
                ),
                entities=[
                ],
                replies=MessageReplies(
                    replies=0,
                    replies_pts=173,
                    comments=False,
                    recent_repliers=[
                    ],
                    channel_id=None,
                    max_id=None,
                    read_max_id=None
                ),
            ),
            pts=173,
            pts_count=1
        ),
    ],
    date=datetime.datetime(2021, 1, 23, 23, 25, 59, tzinfo=datetime.timezone.utc),
    seq=0
)
154 d
```

</details>

Note that it's not until "c" that the `pts` gap can be detected. One possible solution is to always get difference when we receive a forward in a channel. This is of course incredibly stupid, and likely a server bug, because **the server be sending all new messages**. It has always done that, and for some reason they have changed this.

Update handling is a mess in Telethon, and fixing this would require **serious structural changes**. I'm not even done implementing it in [`grammers`](https://github.com/lonami/grammers), which is a clean implementation and where I can develop with Rust's safety. Channels still need care. Doing this in Telethon will require some serious work, and probably the Rust implementation to be complete first, so that I can get a good grasp at how this should be done in a proper way.

TL;DR; it would be great if someone could report this issue to the server guys because otherwise it's going to take me a while to fix it. Contributions to either library welcome! My free time is extremely limited, but the sooner we can fix this the better for everyone. At the very least, it would be great to hear the rationale behind this change, and if it's intended or just a bug.

**Comment by uzlov** [comment]
Thx for your explaining @Lonami

Partially working solution here
(I don't tested "album" event, because I'm using my custom solution for manage by "grouped_id")
https://github.com/uzlov/Telethon/tree/fix-update-new-messages

Actually, it's a the dirty hack of Telethon :(
Sorry for my code, I have not enough experience in python and Telegram

**Comment by uzlov** [comment]
I guees, issue can be closed. Looks like, Telegram is fixed issue (all messages is existing)!

**Comment by Lonami** [comment]
@L8KKqTNnge or @SharkLabDev can you confirm?

**Comment by SharkLabDev** [comment]
Yes. Fixed
